doc/lsst.ip.diffim/index.rst

```diff
@@ -15,14 +34,72 @@ Using lsst.ip.diffim
 
 .. toctree linking to topics related to using the module's APIs.
 
-.. toctree::
+.. toctree
    :maxdepth: 1
 
-.. toctree::
+.. toctree
    :maxdepth: 1
 
-   code_notes
-   AL_implementation
+Production pipelines
+--------------------
+
+This package is generally being used in production pipelines like ``lsst.ap.pipe`` and ``lsst.drp.pipe``, designed for processing large collections of survey data.
+An extract of an example pipeline definition file can be found below, where tasks from ``lsst.ip.diffim`` are being used and configured explicitely.
+
+.. code-block:: yaml
+
+   imports:
+   - location: $AP_PIPE_DIR/pipelines/_ingredients/ProcessCcd.yaml
+   parameters:
+      coaddName: goodSeeing
+   tasks:
+      retrieveTemplate:  # For multi-tract difference imaging
+         class: lsst.ip.diffim.getTemplate.GetTemplateTask
+         config:
+            connections.coaddName: parameters.coaddName
+      subtractImages:
+         class: lsst.ip.diffim.subtractImages.AlardLuptonSubtractTask
+         config:
+            connections.coaddName: parameters.coaddName
+            doApplyExternalCalibrations: False
+      detectAndMeasure:
+         class: lsst.ip.diffim.detectAndMeasure.DetectAndMeasureTask
+         config:
+            connections.coaddName: parameters.coaddName
+            doSkySources: True
+   ...
+
+Single frame processing
+-----------------------
+
+Single image frame usage can be carried out by importing the ``lsst.ip.diffim`` as a library of tasks, as the following example shows:
+
+.. code-block:: python
+
+   from lsst.ip.diffim import AlardLuptonSubtractConfig 
+   from lsst.ip.diffim import AlardLuptonSubtractTask
```

Comment:
Instead I'd do `from lsst.ip.diffim import subtractImages`; then you can use that namespace below.